pipeline_dp/pipeline_backend.py

```diff
@@ -246,14 +288,23 @@ def map_with_side_inputs(self,
                              side_input_cols,
                              stage_name: str = None):
         side_inputs = [
-            beam.pvalue.AsList(side_input_col)
+            beam.pvalue.AsSingleton(side_input_col)
```

Comment:
is it more efficient in terms of speed? It makes more sense to use `AsList` because it use more memory-efficient.

Reply:
I don't think there are any difference in efficient. This is simplification, in all our cases we have 1 element collection (and we always can get 1-element collection with `to_list`) and I found it's confusing for new people, that `fn` takes a list of 1-element